pkg/services/ngalert/models/receivers.go

```diff
@@ -206,40 +289,127 @@ func (integration *Integration) Clone() Integration {
 		Name:                  integration.Name,
 		Config:                integration.Config.Clone(),
 		DisableResolveMessage: integration.DisableResolveMessage,
-		Settings:              maps.Clone(integration.Settings),
+		Settings:              mapClone(integration.Settings),
 		SecureSettings:        maps.Clone(integration.SecureSettings),
 	}
 }
 
+func mapClone(m map[string]any) map[string]any {
```

Comment:
this is the most questionable part. I am not a fan of this implementation but it should be ok considering the types we use in integration settings. A simpler alternative would be to use marshalling to clone the map